hooks/react-client/useBottomScrollListener.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+
+type CallbackFunction = () => void;
+
+const useBottomScrollListener = (
+  callback: CallbackFunction,
+  debounceTime = 300
+) => {
+  const [bottomReached, setBottomReached] = useState(false);
+  let timeoutId: NodeJS.Timeout | null = null;
+
+  const handleScroll = () => {
+    const scrollTop = document.documentElement.scrollTop;
+    const windowHeight = window.innerHeight;
+    const height = document.documentElement.scrollHeight;
+
+    const bottomOfWindow = Math.ceil(scrollTop + windowHeight) >= height;
+
+    if (bottomOfWindow) {
```

Comment:
I think you should use a proper debounce function. You can ask Copilot to ask you a debounce function and then just put it under utils. Cleaner and easier to understand than this.

Reply:
```tsx
type DebounceFunction = (...args: any[]) => void;

function debounce<T extends DebounceFunction>(
  func: T,
  delay: number
): (...args: Parameters<T>) => void {
  let timeoutId: NodeJS.Timeout;

  return (...args: Parameters<T>) => {
    clearTimeout(timeoutId);

    timeoutId = setTimeout(() => {
      func(...args);
    }, delay);
  };
}

export default debounce;
```

Reply:
in which file would you put this?

Reply:
Within `/utils/debounce.ts` maybe.